ibis/sql/transforms.py

```diff
@@ -26,27 +28,18 @@ def type(self):
         return 'exists'
 
 
-class ExistsSubquery(ir.Node):
-
-    """
-    Helper class
-    """
-
-    def __init__(self, foreign_table, predicates):
-        self.foreign_table = foreign_table
-        self.predicates = predicates
-        ir.Node.__init__(self, [foreign_table, predicates])
+class ExistsSubquery(ops.Node):
+    """Helper class"""
+    foreign_table = Arg(rlz.noop)
+    predicates = Arg(rlz.noop)
 
     def output_type(self):
         return ExistsExpr
 
 
-class NotExistsSubquery(ir.Node):
-
-    def __init__(self, foreign_table, predicates):
-        self.foreign_table = foreign_table
-        self.predicates = predicates
-        ir.Node.__init__(self, [foreign_table, predicates])
+class NotExistsSubquery(ops.Node):
+    foreign_table = Arg(rlz.noop)
+    predicates = Arg(rlz.noop)
```

Comment:
Nice to see this finally getting rolled into the rules system.

Reply:
We need to revisit all Args with `rlz.noop` :)